### ISSUE TITLE
Debug branch: Soundness bug for NIRA formula

### ISSUE THREAD

**Issue by muchang** [Z3Prover/z3#3375]
Hi,
For this formula:
```
(declare-fun v () Real) 
(declare-fun d () Real) 
(declare-fun b () Real) 
(declare-fun m () Real) 
(declare-fun t2uscore0dollarskuscore0 () Real) 
(declare-fun z () Real) 
(assert (not (exists ((ts2uscore0 Real)) 
    (=> (or (and (and (= (<= ts2uscore0 ts2uscore0 t2uscore0dollarskuscore0) 
    (>= v 0)) (>= t2uscore0dollarskuscore0 172)) (<= (- (div d d)) 0)) (<= z m)) 
    (<= (/ 0 v) d))))) 
(check-sat)
```
Z3 debug branch  gives an incorrect answer:
```
$ z3debug/build/z3 small.smt2 
unsat
$ z3/build/z3 small.smt2 
sat
$ cat small.smt2 
(declare-fun v () Real) 
(declare-fun d () Real) 
(declare-fun b () Real) 
(declare-fun m () Real) 
(declare-fun t2uscore0dollarskuscore0 () Real) 
(declare-fun z () Real) 
(assert (not (exists ((ts2uscore0 Real)) 
    (=> (or (and (and (= (<= ts2uscore0 ts2uscore0 t2uscore0dollarskuscore0) 
    (>= v 0)) (>= t2uscore0dollarskuscore0 172)) (<= (- (div d d)) 0)) (<= z m)) 
    (<= (/ 0 v) d))))) 
(check-sat)
```

OS: Ubuntu 18.04
Commit: aa119d1

**Comment by levnach** [comment]
I am not sure how to fix it. The scenario is as following. lar_solver::add_equality() is called with two variables that have different values at the moment. This triggers setting the lar_solver to infeasible, and also set m_infeasible_column equal to the newly created column. Then the explanation is taken from this column, but the corresponding lemma is created empty - just (check-sat).

**Comment by levnach** [comment]
Closed by 62bd8519ecc1f002d525f5c9a3776640effba1dd
